src/rover/main.cpp

```diff
@@ -0,0 +1,174 @@
+//#define DEBUG
+#define TIME
+
+#include "camera.hpp"
+#include "../network/network.hpp"
+
+#include <cstddef>
+#include <iostream>
+#include <vector>
+#include <cstring>
+#include <cstdlib>
+#ifdef TIME
+	#include <chrono> // Time keeping from stdlib
+	using namespace std::chrono;
+#endif
+
+
+const unsigned int CAMERA_MESSAGE_SIZE = 65000;
+
+int main() {
+	/*
+	 * INITILIZATION: Rover Objects
+	 * OPTIMIZE: Consider making a rover class with an iteratable list of cameras
+	 */
+	unsigned int frame_counter = 0;
+	// Open all cameras
+	camera::CaptureSession session0;
+	camera::CaptureSession session1;
+	camera::CaptureSession session2;
+	camera::CaptureSession session3;
+	camera::Error err0 = camera::open(&session0, "/dev/video0", 1280, 720);
+	camera::Error err1 = camera::open(&session1, "/dev/video1", 1280, 720);
```

Comment:
I would either:
1. Use a single error variable;
2. Put each open call in a local block:
```
{
    camera::Error err = camera::open(&session1, "/dev/video1", 1280, 720);
    if (err != camera::Error::OK) { /* handle error */ }
}
```
Either way, make sure to check these errors after they occur!

And since you are just duplicating the feed from the first camera session for all 4 streams, just open one camera session up here.